### PR TITLE
`sigstore verify`: support multiple inputs

### LINKED ISSUE #127: Infer signature and certificate paths from `sigstore verify`'s input
Right now, `sigstore verify` needs to be used like this:

```bash
sigstore verify --certificate foo.crt --signature foo.sig foo
```

This has a handful of downsides:

* It violates one of the Unix tool idioms, which is that flags should never be mandatory
* It fundamentally constrains the CLI to verifying one file at a time, since `--certificate` and `--signature` can only be passed once (in theory they could be made plural, but this results in poor UX where a user must either accept quadratic verification times *or* ensure that their flag order is exactly right)

One potential solution to this is to do the converse of what we've done with `sigstore sign`, and make `--certificate` and `--signature` optional. When absent, multiple inputs will be allowed, with the following lookup scheme:

```
input.ext -> [input.ext.crt, input.ext.sig]
```

This makes the happy path _much_ cleaner:

```bash
# automatically discovers foo.tar.gz.{crt,sig}
sigstore verify foo.tar.gz

# automatically discovers .crt and .sig for each of {foo,bar}.tar.gz
sigstore verify foo.tar.gz bar.tar.gz
```

Thoughts @di?

### PR DESCRIPTION
**Pull request by woodruffw**
Closes #127.

### CODE REVIEW

**Comment by woodruffw** [comment]
I'll also use this PR to update the README.

File: sigstore/_cli.py [hunk 1]
```diff
@@ -362,6 +367,38 @@ def _sign(args: argparse.Namespace) -> None:
...
+
+        missing = []
+        if not sig.is_file():
+            missing.append(str(sig))
+        if not cert.is_file():
+            missing.append(str(cert))
+
+        if missing:
+            args._parser.error(
+                f"Missing requirements for {(file)}: {', '.join(missing)}"
```
Comment (di) [inline sigstore/_cli.py:397]:
```suggestion
                f"Missing verification materials for {(file)}: {', '.join(missing)}"
```

File: sigstore/_cli.py [hunk 2]
```diff
@@ -362,6 +367,38 @@ def _sign(args: argparse.Namespace) -> None:
 
 
 def _verify(args: argparse.Namespace) -> None:
+    # Fail if `--certificate` or `--signature` is specified and we have more than one input.
+    if (args.certificate or args.signature) and len(args.files) > 1:
+        args._parser.error(
+            "--certificate and --signature can only be used with a single input"
```
Comment (di) [inline sigstore/_cli.py:373]:
```suggestion
            "--certificate and --signature can only be used with a single input file"
```

**Review by di — DISMISSED** [review]

**Review by di — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -215,6 +216,55 @@ $ python -m sigstore sign --identity-token YOUR-LONG-JWT-HERE foo.txt
 Note that passing a custom identity token does not circumvent Fulcio's requirements,
 namely the Fulcio's supported identity providers and the claims expected within the token.
 
+### Verifying against a signature and certificate
+
+By default, `sigstore verify` will attempt to find a `foo.txt.sig` and `foo.txt.crt` adjacent
+to the file being verified:
```
Comment (di) [inline README.md:222]:
```suggestion
By default, `sigstore verify` will attempt to find a `<filename>.sig` and `<filename>.crt` in the
same directory as the file being verified:
```

**Review by tetsuo-cpp — APPROVED** [review]
LGTM!

**Pull request merged**